### DISCUSSION TITLE
¡¡Praise!! (& Ideas).. For Seal, @JunkFood02, and Contributors (that's you!); Ideas list-comprehensive; implementations, improvements upon, & combining features and functions found in related apps (for reference:) NewPipe, DVD, Spotiflyer, *SponsorBlock, *RYD API (Return Youtube Dislikes). [*used in Vanced]

### DISCUSSION THREAD

**Discussion by lpstandardrp** [JunkFood02/Seal#119] [Ideas]
Been using DVD <3 
I wanted more functions (playlist downloads,, subtitles, description,, metadata, etc.) And preferably a GUI because it has been time consuming and complicated for my current skill level to track down, understand, and try to execute [yt-dlp options](https://github.com/yt-dlp/yt-dlp#usage-and-options) within the DVD app. 
Fortunately, I saw this [issue](https://github.com/yausername/dvd/issues/60#issuecomment-1153888544)... 
> dvd GUI is designed to download single video in a time. Parsing playlist json is also not supported in youtubedl-android currently. But you can try [Seal](https://github.com/JunkFood02/Seal), another yt-dlp GUI wrapper written by me.

...which introduced me to Seal! Hopefully, Seal will resolve these struggles.  I imagine it has great potential for becoming (or becoming even more of) an invaluable and majorly impactful and valuable app!! I am excited and eager to watch and see this app grow!!

Each "downloader" app I have experienced is invaluable for specific uses (Newpipe, DVD, Spotiflyer); beyond those uses is where I hope Seal comes to the rescue!! Namely,, a GUI (or at least common commands and examples included in-app (not a hyperlink); for example in a "help" or "manual" or "tutorial" document contained within the app) and options for 

- Audio, Video, Audio *and* Video files, Subtitles (including embedded into video option, .srt file, etc. )
-  codecs, containers(IIRC), formats, etc. (m4a, mp3, opus, webm, etc.)
- Optional metadata or data- 
 - downloaded as separate file
 - downloaded and included in audio/video/___ file (as metadata, if I understand metadata correctly), ...
- Some options that I would love to see:
 - Description
 - Comments, 
 - Chapters and Markers timeline
 - Channel Playlist (if applicable)
  - [#] within channel video playlist (i.e. (a made up example): video [#]2 of 5 in;  [YouTube Channel:] Seal App;  Seal App's Video Playlist: 'Seal (app)(Android)(yt-dlp)(FLOSS) - Intro tutorials getting started (5 videos)')
  - Include Playlist Track # in filename
- SponsorBlock (related) - sponsor.ajay.app API aka "Sponsorblock Settings"  (as found in Vanced)
 - skip/remove ads and sponsor segments in downloaded file (using data from sponsor.ajay.app API aka "Sponsorblock Settings"  (as found in Vanced))
 - Include ads /sponsor segments in download
  - include markors/chapters in file and/or metadata*  
   - *note: *is this possible? does it exist? If so,, what file formats|codecs available via yt-dlp include "markers"|"chapters" as optional file data|metadata?  
   - download into separate file: video timeline, Chapters, Markers, including Sponsors and Ads segments (SponsorBlock data via API)
- closed captioning, subtitles, which version: official uploaded subtitles, "auto-generated", etc.

Some that may be of value
- views, likes, dislikes* 
 - *Might need to access or use for reference: "RYD API" aka "Return youtube dislike settings" (as found in Vanced)(R.I.P.))
 - (stats:) views/likes/dislikes ratio 
- similar video titles - urls, upload date, views and likes/dislikes ratio 

Sending Thanks up the wazoo, and ¡praise! For Seal, @JunkFood02, and Contributors (that's you!), and... many times!!

Thank y'alls <3

**Comment by JunkFood02** [comment]
Thanks for the like 😉
For the request for more built-in options, I'll address this in a new discussion

**Reply by lpstandardrp** [reply]
> Thanks for the like wink For the request for more built-in options, I'll address this in a new discussion

@JunkFood02 ~ Awesome. Thank YOU!! Stay rad!
(Please Lmk if you think tagging (like I did tagging you here @JunkFood02) for a thank you response in Github is generally considered: 
[] 1) (Yes) better to tag user (appreciated, helpful, valuable) or 
[] 2) (No) not appropriate, valuable enough, etc. (I.e. causes extra work, is unnecessaty, distracting, etc.) 
[] 3 (i.e. Other, neither Yes or No, both Yes and No, etc.)

Or, if you don't know, what was the first thought that popped in your head, or your first gut instinct response?
[]41 (Yes)
[]42 (No)
[]43 (Other, ...)

Cheers! Best Regards~
